### PR TITLE
feat(express rule): detect missing usage within same file

### PR DESCRIPTION
**Pull request by cfabianski**
## Description

Support negative rule.
Introduced a new type of trigger `trigger: absence` which requires a `trigger_rule_on_presence_of`

The way it works is, we check for the presence of a detection id (in this case `javascript_express_helmet_express_init` which is to check that express is initialized within the file), and if that one exists but we couldn't find any detection for the rule itself, we will raise the failure.

## Checklist

- [ ] I've added test coverage that shows my fix or feature works as expected.
- [ ] I've updated or added documentation if required.
- [ ] I've included usage information in the description if CLI behavior was updated or added.
- [ ] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: pkg/commands/process/settings/rules/javascript/express/helmet.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        detection: javascript_express_helmet_express_init
+      - variable: FUNCTION_CALL
+        detection: javascript_express_helmet_configuration
+trigger_rule_on: javascript_express_helmet_express_init
+trigger_rule_on_absent: javascript_express_helmet
+auxiliary:
+  - id: javascript_express_helmet_express_init
+    patterns:
+      - express()
+  # FIXME: we don't support `const helmet = require("helmet")` yet!
```
Comment (elsapet) [inline pkg/commands/process/settings/rules/javascript/express/helmet.yml]: The same with import statements

```javascript
import { helmet as customName } from "helmet"
```

File: pkg/commands/process/settings/rules/javascript/express/helmet.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+patterns:
+  - pattern: |
+      $<EXPRESS>.use($<FUNCTION_CALL>)
+    filters:
+      - variable: EXPRESS
+        detection: javascript_express_helmet_express_init
+      - variable: FUNCTION_CALL
+        detection: javascript_express_helmet_configuration
+trigger_rule_on: javascript_express_helmet_express_init
+trigger_rule_on_absent: javascript_express_helmet
```
Comment (elsapet) [inline pkg/commands/process/settings/rules/javascript/express/helmet.yml]:
```suggestion
trigger_rule_on_absence: javascript_express_helmet
```

File: pkg/report/output/security/security.go [hunk 3]
```diff
@@ -134,6 +135,7 @@ func evaluateRules(
 
 		if len(rs) > 0 {
 			jsonRes, err := json.Marshal(rs)
+			log.Error().Msgf("jsonRes: %s", jsonRes)
```
Comment (elsapet) [inline pkg/report/output/security/security.go]: Do we want to remove this?

Comment (cfabianski) [reply pkg/report/output/security/security.go]: Yeah i need to clean it up a little bit still

**Review by elsapet — APPROVED** [review]
Not sure where's best to keep track of the pending documentation changes (how to use the new trigger attributes)?

**Pull request merged**